src/core/audio/AudioSoundIo.cpp

```diff
@@ -0,0 +1,405 @@
+/*
+ * AudioSoundIo.cpp - device-class that performs PCM-output via libsoundio
+ *
+ * Copyright (c) 2015 Andrew Kelley <superjoe30@gmail.com>
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "AudioSoundIo.h"
+
+#ifdef LMMS_HAVE_SOUNDIO
+
+#include <QLabel>
+#include <QLineEdit>
+
+#include "Engine.h"
+#include "debug.h"
+#include "ConfigManager.h"
+#include "gui_templates.h"
+#include "templates.h"
+#include "ComboBox.h"
+#include "LcdSpinBox.h"
+
+AudioSoundIo::AudioSoundIo( bool & out_successful, Mixer * _mixer ) :
+    AudioDevice( tLimit<ch_cnt_t>(
+        ConfigManager::inst()->value( "audiosoundio", "channels" ).toInt(), DEFAULT_CHANNELS, SURROUND_CHANNELS ),
+                                _mixer )
+{
+    out_successful = false;
+    m_soundio = NULL;
+    m_outstream = NULL;
+    m_disconnect_err = 0;
+    m_outBufFrameIndex = 0;
+    m_outBufFramesTotal = 0;
+
+    m_soundio = soundio_create();
+    if (!m_soundio) {
```

Comment:
Follow the brace conventions used in the rest of lmms, i.e.
```c++
if (!m_soundio)
{
...
}
```

Just marking this one location, but it needs to be changed throughout this file.